noir_stdlib/src/hash/poseidon/bn254.nr

```diff
@@ -2,93 +2,14 @@
 mod perm;
 mod consts;
 
-use crate::hash::poseidon::PoseidonConfig;
-use crate::hash::poseidon::apply_matrix;
-// Optimised permutation for this particular field; uses hardcoded rf and rp values,
-// which should agree with those in pos_conf.
-#[field(bn254)]
-pub fn permute<M, N, O>(pos_conf: PoseidonConfig<M, N>, mut state: [Field; O]) -> [Field; O] {
```

Comment:
Do we not want a bn254 permute anymore?

Reply:
Oh wait did you remove this cause we actually require bn254 on the all the methods in `noir_stdlib/src/hash/poseidon/bn254/perm.nr`? Ignore this then

Reply:
I removed this as this was just exactly the same as the generic permute but had hardcoded constants (something that we get for free if the config is a constant). I've then removed this and kept the generic version.